addons/website/models/website_visitor.py

```diff
@@ -341,24 +373,18 @@ def _update_visitor_timezone(self, timezone):
         self.env.cr.execute(query, (timezone, self.id))
 
     def _update_visitor_last_visit(self):
-        """ We need to do this part here to avoid concurrent updates error. """
-        try:
-            with self.env.cr.savepoint():
-                query_lock = "SELECT * FROM website_visitor where id = %s FOR NO KEY UPDATE NOWAIT"
-                self.env.cr.execute(query_lock, (self.id,), log_exceptions=False)
-
-                date_now = datetime.now()
-                query = "UPDATE website_visitor SET "
-                if self.last_connection_datetime < (date_now - timedelta(hours=8)):
-                    query += "visit_count = visit_count + 1,"
-                query += """
-                    active = True,
-                    last_connection_datetime = %s
-                    WHERE id = %s
-                """
-                self.env.cr.execute(query, (date_now, self.id), log_exceptions=False)
-        except Exception:
-            pass
+        date_now = datetime.now()
+        query = "UPDATE website_visitor SET "
+        if self.last_connection_datetime < (date_now - timedelta(hours=8)):
+            query += "visit_count = visit_count + 1,"
+        query += """
```

Comment:
Don't use string concatenation, this makes code hard to read and understand. I think you can achieve better results preparing optional query parts with meaningful name (not ``query`` but more like ``update_optional_visit``), and doing something like a final fstring that can be read directly. Otherwise it is a ping-pong-like code.